docs/authentication.md

```diff
@@ -0,0 +1,130 @@
+
+Also, note that we use the term `signature` here to specify a byte array used for authentication.
+
+## Authentication methods
+
+ScalarDL supports two authentication methods: Digital Signatures and HMAC.
+The methods have advantages and disadvantages, as shown below, but Byzantine fault detection capability won't be sacrificed either way.
+
+### Digital Signatures
+* Advantages
+  * Asset records and asset proofs have non-repudiation property. Specifically, a digital signature attached to a request is stored with the corresponding asset records produced by the request so that the records have non-repudiation property, i.e., we can ensure that the records are created by the owner of the private key that signed the request. Moreover, digital signatures attached to asset proofs returned to a client as the result of execution ensure that the proofs are created by Ledger and Auditor, respectively. If a client (application) keeps the proofs, the client can verify results with the proofs as necessary.
+* Disadvantages
+  * Digital Signatures are very slow. They will add non-negligible performance overhead in exchange for the above benefits.
+
+### HMAC
+* Advantages
+    * HMAC is way faster than digital signatures.
+* Disadvantages
+    * Asset records and asset proofs do not have non-repudiation property.
+
+### Which should I use?
+
+If you do not require non-repudiation property, you should always use HMAC.
+Technically, you could mix authentication methods, e.g., use digital signatures for client authentication and use HMAC for Ledger/Auditor authentication, but it is very confusing; thus, ScalarDL prohibits such usage.
+
+Note that we plan to update ScalarDL to use only HMAC for Ledger and Auditor authentication for better performance. Similarly, we plan to unbundle Ledger (Auditor) authentication from how we sign asset proofs. With the above changes, we will be able to return digitally-signed asset proofs while using HMAC authentication between Ledger and Auditor.
+
+## Configure
+
+This section explains what variables you need to configure to use the authentication of ScalarDL properly. For the detail of each variable, please also check the [Javadoc](TODO).
+
+### Digital Signatures
+
+* Client authentication
+    * Client-side properties
+        * `scalar.dl.client.auditor.enabled` (set true if you use Auditor)
+        * `scalar.dl.client.authentication_method` (set "digital-signature")
+        * `scalar.dl.client.entity.id` (or `scalar.dl.client.cert_holder_id`, which is deprecated.)
+            * It is used for identifying a client.
+        * `scalar.dl.client.entity.identity.digital_signature.cert_path` or `scalar.dl.client.entity.identity.digital_signature.cert_pem` (or `scalar.dl.client.cert_pem` or `scalar.dl.client.cert_path`, which are deprecated.)
```

Comment:
To be consistent order with the other items, I suggest the order `scalar.dl.client.entity...cert_pem` then `scalar.dl.client.entity...cert_path`, rather than `scalar.dl.client.entity...cert_path` then `scalar.dl.client.entity...cert_pem`

Reply:
Good catch! Thanks. Fixed in [421a7b0](https://github.com/scalar-labs/scalardl/pull/45/commits/421a7b0118977c6786d0212dfdfcc61cdfe5e910).
The others start with `pem` so made it consistent with them.